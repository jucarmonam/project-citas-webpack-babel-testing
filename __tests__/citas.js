import Citas from "../js/classes/Citas";

describe("Probar la clase de citas", () => {
  const citas = new Citas();

  const id = Date.now();

  test("Agregar una nueva cita", () => {
    const citaObj = {
      id,
      mascota: "Hook",
      propietario: "Juan",
      telefono: "324234",
      fecha: "10-01-2021",
      hora: "10:30",
      sintomas: "Solo duerme",
    };

    citas.agregarCita(citaObj);

    //Prueba

    expect(citas).toMatchSnapshot();
  });

  test("actualizar cita", () => {
    const citaActualizada = {
      id,
      mascota: "Nuevo Nombre",
      propietario: "Juan",
      telefono: "324234",
      fecha: "10-01-2021",
      hora: "10:30",
      sintomas: "Solo duerme",
    };

    citas.editarCita(citaActualizada);

    expect(citas).toMatchSnapshot();
  });

  test("Eliminar cita", () => {
    citas.eliminarCita(id);

    expect(citas).toMatchSnapshot();
  });
});
