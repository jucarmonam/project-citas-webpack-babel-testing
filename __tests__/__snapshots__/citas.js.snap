// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-01-2021",
      "hora": "10:30",
      "id": 1610644605294,
      "mascota": "Hook",
      "propietario": "Juan",
      "sintomas": "Solo duerme",
      "telefono": "324234",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;

exports[`Probar la clase de citas actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-01-2021",
      "hora": "10:30",
      "id": 1610644605294,
      "mascota": "Nuevo Nombre",
      "propietario": "Juan",
      "sintomas": "Solo duerme",
      "telefono": "324234",
    },
  ],
}
`;
